tesseract_environment/src/core/environment.cpp

```diff
@@ -161,227 +162,159 @@ bool Environment::isInitialized() const { return initialized_; }
 
 int Environment::getRevision() const { return revision_; }
 
-const Commands& Environment::getCommandHistory() const { return commands_; }
+Commands Environment::getCommandHistory() const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Should we make this a shared mutex?

Reply:
I think that would be good. I will do this in a separate commit or PR after the clean up.

Reply:
This has been added in a single commit.